### PR TITLE
Apimikov/multiplayers

### PR DESCRIPTION
**Pull request by dnapi**
The basic version of 4 player + 2 ball game is implemented . 
It is game 2 vs 2. For testing purpose max score is 5.   Control buttons are  WS RF OL ArrowUp/Down. 
Also partially ~~un~~successful attempt to implement Pause (SPACE) and termination (ESC)  was done, but work is still in the progress. 
Code of main.js script was refactored to make it simpler to extend the case of 2 players to 4 players. 1 ball  to 2 balls.

I do not know why test.py was modified: new class added `class Migration(migrations.Migration):   ....`

### CODE REVIEW

**Review by DaveeHorvath — COMMENTED** [review]
one critical issue i left in and am just now realising is that the creation of the threejs scenes, cameras and similar should happen once, since we want to make it run multiple times, it would break (probably)

File: frontend/src/main.js [hunk 1]
```diff
@@ -3,10 +3,71 @@ import * as THREE from 'three';
 import { FontLoader } from 'three/addons/loaders/FontLoader.js';
 import { TextGeometry } from 'three/addons/geometries/TextGeometry.js';
 
+
```
Comment (DaveeHorvath) [inline frontend/src/main.js:6]: good change, but some of these should be changeable, maybe in the gamestart parameters

File: frontend/src/main.js [hunk 2]
```diff
@@ -16,10 +77,12 @@ function getRandom(min, max) {
...
-    let gameResult = "Game is not finished yet"
-    let maxScore = 11
-    let hight = 20;
-    let width = 40;
+    let gameResult = "Game is not finished yet";
+    let isPaused = false;
+    const maxScore = 5;
+    const playerSpeed = 17;  //12
+    
+    // create a scene and camera
```
Comment (DaveeHorvath) [inline frontend/src/main.js:85]: these could be moved outside, maybe have some initialisation to them in a static like manner=> if they havent been created create them

**Review by DaveeHorvath — APPROVED** [review]
looks good, i left a couple small requests in there, they might solve the issues down the line, can proceed to pull, they can be addressed in the next pr

File: frontend/src/main.js (hunk 2 above)
Comment (nikkxll) [reply frontend/src/main.js:85]: We might also want to move it to the state json that we elaborated together in order to add it to the settings later.

**Comment by nikkxll** [comment]
Good job done here!
There are a couple of recommendations:
- Regarding sticky keys I found out this thread, could be useful to try add time intervals and check if button released or not. Here is the link: https://stackoverflow.com/questions/10949369/javascript-stuck-keys-unregistered-keyup-event
- Might be better to use switch-case instead of if-else constructions (faster and cleaner)
- We might also think about which buttons to use for the game for convenience sake
- Found some problems with pause and ESC. Clicking SPACE second time just close the game window, weird behaviour when pressing ESC after SPACE and so on.

**Pull request merged**